rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -522,6 +551,234 @@ void save_callsFederationSupportSave() {
+
+            bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeMainNetConstants)
+                .withEventLogger(bridgeEventLogger)
+                .withProvider(bridgeStorageProvider)
+                .withActivations(activationsAfterForks)
+                .withFederationSupport(federationSupport)
+                .withFeePerKbSupport(feePerKbSupport)
+                .withExecutionBlock(rskExecutionBlock)
+                .build();
+        }
+
+        @Test
+        void processSvpFundTransactionWithoutSignatures_whenProposedFederationDoesNotExist_throwsIllegalStateException() {
+            // arrange
+            when(federationSupport.getProposedFederation()).thenReturn(Optional.empty());
+
+            // act & assert
+            IllegalStateException exception = assertThrows(IllegalStateException.class,
+                () -> bridgeSupport.processSvpFundTransactionUnsigned(rskTx));
+
+            String expectedMessage = "Proposed federation should be present when processing SVP fund transaction.";
+            assertEquals(expectedMessage, exception.getMessage());
+        }
+
+        @Test
+        void processSvpFundTransactionWithoutSignatures_createsExpectedTransactionAndSavesTheHashInStorageEntryAndPerformsPegoutActions() throws Exception {
+            // act
+            bridgeSupport.processSvpFundTransactionUnsigned(rskTx);
+            bridgeStorageProvider.save(); // to save the tx sig hash
+
+            // assert
+            assertSvpFundTxHashUnsignedWasSavedInStorage();
+            assertSvpReleaseWasSettled();
+            assertSvpFundTransactionHasExpectedInputsAndOutputs();
+        }
+
+        private void assertSvpFundTxHashUnsignedWasSavedInStorage() {
+            Optional<Sha256Hash> svpFundTransactionHashUnsignedOpt = bridgeStorageProvider.getSvpFundTxHashUnsigned();
+            assertTrue(svpFundTransactionHashUnsignedOpt.isPresent());
+
+            svpFundTransactionHashUnsigned = svpFundTransactionHashUnsignedOpt.get();
+        }
+
+        private void assertSvpReleaseWasSettled() throws IOException {
+            PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = bridgeStorageProvider.getPegoutsWaitingForConfirmations();
+            assertPegoutWasAddedToPegoutsWaitingForConfirmations(pegoutsWaitingForConfirmations, svpFundTransactionHashUnsigned, rskTxHash);
+            getSvpFundTransactionFromPegoutsMap(pegoutsWaitingForConfirmations);
+
+            assertPegoutTxSigHashWasSaved(svpFundTransactionUnsigned);
+            assertLogReleaseRequested(rskTxHash, svpFundTransactionHashUnsigned, bridgeMainNetConstants.getSpendableValueFromProposedFederation());
+            assertLogPegoutTransactionCreated(svpFundTransactionUnsigned);
+        }
+
+        private void getSvpFundTransactionFromPegoutsMap(PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations) {
```

Comment:
This is a bit strange. The method is called `get` but it's void

Reply:
good one, thanks